tests/test_mock_temperature_sensor.py

```diff
@@ -0,0 +1,81 @@
+# This file is part of ts_ess_sensors.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+import asyncio
+import unittest
+
+from lsst.ts.ess.sensors.mock.mock_temperature_sensor import (
+    MockTemperatureSensor,
+    MIN_TEMP,
+    MAX_TEMP,
+)
+
+
+class MockTestCase(unittest.IsolatedAsyncioTestCase):
+    async def test_read_instrument(self):
+        num_channels = 4
+        name = "MockSensor"
+        ess_sensor = MockTemperatureSensor(name, num_channels)
+        ess_sensor.terminator = "\r\n"
+        loop = asyncio.get_event_loop()
+        device_name, err, resp = await loop.run_in_executor(None, ess_sensor.readline)
+        self.assertEqual(name, device_name)
+        resp = resp.strip(ess_sensor.terminator)
+        data = resp.split(",")
+        for i in range(0, num_channels):
+            data_item = data[i].split("=")
+            self.assertTrue(f"C{i:02d}", data_item[0])
+            self.assertTrue(MIN_TEMP <= float(data_item[1]) <= MAX_TEMP)
+
+    async def test_read_old_instrument(self):
+        num_channels = 4
+        count_offset = 1
+        name = "MockSensor"
+        ess_sensor = MockTemperatureSensor(name, num_channels, count_offset)
+        ess_sensor.terminator = "\r\n"
+        loop = asyncio.get_event_loop()
+        device_name, err, resp = await loop.run_in_executor(None, ess_sensor.readline)
+        self.assertEqual(name, device_name)
+        resp = resp.strip(ess_sensor.terminator)
+        data = resp.split(",")
+        for i in range(0, num_channels):
+            data_item = data[i].split("=")
+            self.assertTrue(f"C{i + count_offset:02d}", data_item[0])
+            self.assertTrue(MIN_TEMP <= float(data_item[1]) <= MAX_TEMP)
```

Comment:
There is a lot of overlap between this and the previous test method. I strongly suggest you try to consolidate them, perhaps with a "check_read_instrument" that takes a count_offset argument. Are the only valid values for that argument 0 and 1? If not you could call the method with an iterator that tries other values as well.

Reply:
Also this use of <= is code-efficient, but it means a test failure will be hard to interpret. You'll get better error messages if you call assertLessEqual twice (once for the min, once for the max).

Reply:
I adopted the consolidated method and the assertLessEqual suggestions. I'll consider checking for 0 and 1 only values in the mock temperature sensor.